optimum/intel/openvino/configuration.py

```diff
@@ -462,6 +462,11 @@ def post_init(self):
             if self.gptq:
                 raise ValueError("The GPTQ algorithm is not supported for 'mxfp4' weight format")
 
+        if self.weight_format in "nf4" and self.bits != 4:
+            raise ValueError(
+                f"When applying weight compression with 'nf4' weight format the `bits` parameters must be set to 4, but found {self.bits}"
+            )
```

Comment:
I believe AWQ and SE are not supported for NF4, so corresponding checks should be added. GPTQ is supported though. Possibly it could be combined with conditions for mxfp4 above to avoid code duplication.